tests/transforms/test_hamiltonian_expand.py

```diff
@@ -246,7 +247,252 @@ def test_hamiltonian_dif_tensorflow(self):
                     qml.CNOT(wires=[2, 0])
                 qml.expval(H)
 
-            tapes, fn = qml.transforms.hamiltonian_expand(tape)
+            tapes, fn = hamiltonian_expand(tape)
+            res = fn(qml.execute(tapes, dev, qml.gradients.param_shift, interface="tf"))
+
+            assert np.isclose(res, output)
+
+            g = gtape.gradient(res, var)
+            assert np.allclose(list(g[0]) + list(g[1]), output2)
+
+
+with QuantumTape() as s_tape1:
+    qml.PauliX(0)
+    S1 = qml.s_prod(1.5, qml.prod(qml.PauliZ(0), qml.PauliZ(1)))
```

Comment:
I would always be careful with defining things inside a tape context, can you define that outside?

Reply:
I copied these tapes from the ones defined at the beginning of this file, but changing all `Hamiltonian` objects to `Sum` objects.

I think here we intend to use the `Sum` as an operator and as a measurement